src/Core/src/Platform/Android/ApplicationExtensions.cs

```diff
@@ -56,6 +57,9 @@ public static void CreatePlatformWindow(this Activity activity, IApplication app
 					$"or set the LaunchMode to SingleTop on {activity.GetType()}.");
 			}
 
+			var wndProvider = mauiContext.Services?.GetRequiredService<WindowProvider>();
```

Comment:
I don't know if this will worked with multiple levels of scoping. 

The scenario we'd need for `Blazor` is the ability for new scopes to retrieve the correct window Provider

MauiContext.ServiceProvider.CreateScoped.ServiceProvider.CreateScoped.ServiceProvider.GetRequiredService<WindowProvider>();

should be able to get the Window provider that would retrieve the window

Reply:
Do we do/use/support multiple levels of scoping in MAUI? I know we use scoped services already for window-things, so I am not sure if that changes with this specifically.

IF we are using multiple scopes, then I am sure something else would have broken.